### ISSUE TITLE
Rubric questions: allow weightages for sub questions

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/teammates#7224]
Rubric questions allow weightages for choices (i.e. columns)
![image](https://cloud.githubusercontent.com/assets/1673303/25708067/fa3232a6-3117-11e7-88c5-3808e1528643.png)

Suggestion: allow weightages for sub-questions too (i.e. for rows)

Justification: This way, we can calculate an overall score for a student for the entire rubric question.

**Comment by kguo901** [comment]
I will start working on this.

**Comment by damithc** [comment]
@kguo901 I recommend you start with a `d.FirstTimers` issue and then do a few `d.Contributors` issues first. This issue may be too big for you to tackle if you are relatively new to TEAMMATES.

**Comment by kguo901** [comment]
@damithc After looking into this issue more, I agree that I need more familiarity with the code. In the time that I worked on this, I did a bit of layout planning with the new weights (see image below). Whoever wants to work on this issue may feel free to use it.
![teammates - instructo_ - http___localhost_8888_page_instructorfeedbackeditpage](https://cloud.githubusercontent.com/assets/15681154/25750005/70ae3d06-317e-11e7-8ff1-b6e734c9617d.png)

**Comment by adityavhegde** [comment]
Working on this one.

**Comment by SELVAKUMAR-S** [comment]
@adityavhegde , are you on this ? or shall i take this?

**Comment by adityavhegde** [comment]
@SELVAKUMAR-S you can take it.

**Comment by SELVAKUMAR-S** [comment]
thanks, i am taking this!!!

**Comment by SELVAKUMAR-S** [comment]
@damithc , i have question here, On adding weights to the sub questions, how should i calculate the total/average which is to be displayed in the FeedbackResultsPage. And enabling the weight toggle should enable the user to input weight for both columns and rows? or each should have separate toggle?

**Comment by SELVAKUMAR-S** [comment]
hi @damithc can i get insight on this ? please help me out. or this issue is no more a priority

**Comment by damithc** [comment]
@SELVAKUMAR-S yes, it's not a high priority at the moment.

**Comment by Ali96kz** [comment]
Are you meaning that?
![image](https://user-images.githubusercontent.com/15236829/36736202-37da5edc-1c02-11e8-8e82-698c6be7677f.png)

**Comment by damithc** [comment]
> Are you meaning that?

Not really, but now that you mention it, it got me thinking: Perhaps we can simply extend the current `weights` feature to have a weight for each cell rather than each column as it does now? That gives the full flexibility to the user to set any weightage for each cell.

**Comment by Ali96kz** [comment]
Yes.

**Comment by Ali96kz** [comment]
Can't understand how it would be look?

**Comment by damithc** [comment]
> @damithc Can't understand, how it would be look?

It would be similar to your screenshot except there will be only `Weights` checkbox (i.e., keep the current one, no need to add more). When the checkbox is ticked, text boxes will open up in every cell rather than only at the top of the column.

**Comment by Ali96kz** [comment]
I figure out. how it work. I will take this issue.

**Comment by Ali96kz** [comment]
![image](https://user-images.githubusercontent.com/15236829/37156919-96dd59b4-2311-11e8-8009-67c971a14a5f.png)
I must redesign `Response Summary` and `Per Recipient Statistics` table. Will it be normal?
![image](https://user-images.githubusercontent.com/15236829/37157319-b36e091a-2312-11e8-835e-72d1e99950e3.png)

**Comment by damithc** [comment]
A cell can be in this format `80% (2) [1.5]` No need for `Weight`.

**Comment by Ali96kz** [comment]
@damithc Is that normal if we contain a weight for each cell in `List<Double> rubricWeights`?

**Comment by damithc** [comment]
> @damithc Is that normal if we contain a weight for each cell in List<Double> rubricWeights?

You can propose an implementation and we can give feedback on it.

**Comment by Ali96kz** [comment]
Could someone help me with `js`?

**Comment by Ali96kz** [comment]
Completed this issue. I am not well in writing auto-test. Should I just remove it or?

**Comment by jacoblipech** [comment]
Hi @Ali96kz are you still working on this ?

**Comment by sukanta-27** [comment]
@jacoblipech Actually, this issue is a part of my GSoC project.

**Comment by jacoblipech** [comment]
@sukanta-27 okay sure! no worries! 👍

**Comment by sukanta-27** [comment]
@damithc Sir, can we change the name of the issue and the description as we are no more trying to add weights for subquestions but for every individual cell?

**Comment by damithc** [comment]
> @damithc Sir, can we change the name of the issue and the description as we are no more trying to add weights for subquestions but for every individual cell?

Done
